ui/pages/SignTransaction.tsx

```diff
@@ -79,14 +79,13 @@ export default function SignTransaction({
     return undefined
   })
 
-  const areKeyringsUnlocked = useAreKeyringsUnlocked(
-    signerAccountTotal?.signingMethod?.type === "keyring"
-  )
+  const needsKeyrings = signerAccountTotal?.signingMethod?.type === "keyring"
+  const canSign = useAreKeyringsUnlocked(needsKeyrings) || !needsKeyrings
```

Comment:
This nameing here does not feel right. I mean a ledger can also sign. 

The `useAreKeyringsUnlocked` will redirect if the keyring is locked. If it's not of `keyring` type then it shouldn't do that.

Reply:
It's a bit backwards: it should be `const canSign = !needsKeyrings || useAreKeyringsUnlocked(needsKeyrings) ` but it's not allowed because `useAreKeyringsUnlocked` is a hook and must be called unconditionally.

I could invert the logic and write:
```suggestion
  const waitingForKeyrings = !useAreKeyringsUnlocked(needsKeyrings) && needsKeyrings
```

but a more long-term solution would be to move the call `useAreKeyringsUnlocked` to another component that's only rendered when `signerAccountTotal.signingMethod.type === 'keyrings'`

Reply:
Blimey! Let's do this, when we get to this task `update unlock behavior for keyring service to happen after the user clicks "Sign"` https://github.com/tallycash/extension/issues/496